engine/resource/src/providers/provider_archive_mutable.cpp

```diff
@@ -565,7 +565,7 @@ namespace dmResourceProviderArchiveMutable
             dmResource::WriteManifest(manifest_path, archive->m_Manifest);
             dmLogInfo("Wrote manifest to '%s'", manifest_path);
         }
-
+        CreateEntryMap(archive);
```

Comment:
Removing of the old manifest also removes all the hashes which we use as `m_EntryMap` keys. So we need to create a new `m_EntryMap `